be/src/storage/tablet_schema.h

```diff
@@ -85,45 +145,57 @@ class TabletColumn {
     bool is_format_v2_column() const;
 
     int64_t mem_usage() const {
-        int64_t mem_usage =
-                sizeof(TabletColumn) + _col_name.length() + _default_value.length() + _referenced_column.length();
-        for (const auto& col : _sub_columns) {
-            mem_usage += col.mem_usage();
+        int64_t mem_usage = sizeof(TabletColumn) + _col_name.capacity() + default_value().capacity();
+        for (int i = 0; i < subcolumn_count(); i++) {
+            mem_usage += subcolumn(i).mem_usage();
         }
         return mem_usage;
     }
 
 private:
-    int32_t _unique_id;
-    std::string _col_name;
-    FieldType _type;
-    bool _is_key = false;
-    FieldAggregationMethod _aggregation{OLAP_FIELD_AGGREGATION_NONE};
-    bool _is_nullable = false;
-
-    bool _has_default_value = false;
-    std::string _default_value;
+    constexpr static uint8_t kIsKeyShift = 0;
```

Comment:
Why not use mask directly?

Reply:
These constants are used in the method `void _set_flag(uint8_t pos, bool value)` and `bool _check_flag(uint8_t pos)`, IMHO, it's more common and natural for these methods to work with bit position than bit mask.